jdaviz/utils.py

```diff
@@ -494,6 +543,54 @@ def iconverter_flux(x):  # Flux -> Surface Brightness
     ]
 
 
+def _eqv_flux_to_sb_pixel():
+    """
+    Returns an Equivalency between `flux_unit` and `flux_unit`/pix**2. This
+    allows conversion between flux and flux-per-square-pixel surface brightness
+    e.g MJy <> MJy / pix2
+    """
+
+    pix2 = u.pix * u.pix
+
+    # generate an equivalency for each flux type that would need
+    # another equivalency for converting to/from
+    flux_units = [u.MJy, u.erg / (u.s * u.cm**2 * u.Angstrom),
+                  u.ph / (u.Angstrom * u.s * u.cm**2),
+                  u.ph / (u.Hz * u.s * u.cm**2)]
+    equiv = []
+    for flux_unit in flux_units:
+        equiv.append((flux_unit, flux_unit / pix2, lambda x: x, lambda x: x))
+
+    return equiv
+
+
+def _eqv_sb_per_pixel_to_per_angle(flux_unit, scale_factor=1):
+    """
+    Returns an equivalency between `flux_unit` per square pixel and
+    `flux_unit` per solid angle to be able to compare and convert between units
+    like Jy/pix**2 and Jy/sr. The scale factor is assumed to be in steradians,
+    to follow the convention of the PIXAR_SR keyword.
+    Note:
+    To allow conversions between units like 'ph / (Hz s cm2 sr)' and
+    MJy / pix2, which would require this equivalency as well as u.spectral_density,
+    these CAN'T be combined when converting like:
+    equivalencies=u.spectral_density(1 * u.m) + _eqv_sb_per_pixel_to_per_angle(u.Jy)
+    So additional logic is needed to compare units that need both equivalencies
+    (one solution being creating this equivalency for each equivalent flux-type.)
+    """
+    pix2 = u.pix * u.pix
+
+    # the two types of units we want to define a conversion between
+    flux_solid_ang = flux_unit / u.sr
+    flux_sq_pix = flux_unit / pix2
+
+    pix_to_solid_angle_equiv = u.Equivalency([(flux_solid_ang, flux_sq_pix,
```

Comment:
Do you really need `u.Equivalency` here? I don't see it used above.

Reply:
removed in 3156